### PR TITLE
libgpg-error: Enable Aarch32 support for musl

### PR DESCRIPTION
**Pull request by tathougies**
###### Motivation for this change

Following a similar change for x86-64 and musl, we need to copy some files over for arm32 and musl.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/libgpg-error/default.nix [hunk 1]
```diff
@@ -28,8 +28,11 @@ in stdenv.mkDerivation (rec {
     sed '/BUILD_TIMESTAMP=/s/=.*/=1970-01-01T00:01+0000/' -i ./configure
   '' + lib.optionalString (stdenv.hostPlatform.isAarch32 && stdenv.buildPlatform != stdenv.hostPlatform) ''
     ln -s lock-obj-pub.arm-unknown-linux-gnueabi.h src/syscfg/lock-obj-pub.linux-gnueabihf.h
-  '' + lib.optionalString stdenv.hostPlatform.isMusl ''
+  '' + lib.optionalString (stdenv.hostPlatform.isx86_64 && stdenv.hostPlatform.isMusl) ''
```
Comment (c0bw3b) [inline pkgs/development/libraries/libgpg-error/default.nix]: What happens when `isMusl = true` and hostPlatform is not x86-64? (i686, aarch64, ...)

Previously they would match the `optionalString` and the link would have been created, but maybe it was unused.

**Review by c0bw3b — COMMENTED** [review]
Also you commit short message should look like the PR title :)

File: pkgs/development/libraries/libgpg-error/default.nix (hunk 1 above)
Comment (tathougies) [reply pkgs/development/libraries/libgpg-error/default.nix]: It is unused. The configure script only includes the header for the arch being built.

**Comment by tathougies** [comment]
Can I change the commit short message after the fact?

**Comment by c0bw3b** [comment]
To change the commit short message you would need to `git commit --amend` (assuming the commit you want to edit is the last one) and force-push to your `travis/libgpg-error-arm` branch after edition.
It will refresh this PR automatically (we'll see only the new rewritten commit)

**Comment by lheckemann** [comment]
Any reason you used cp rather than ln like the others?

**Comment by tathougies** [comment]
@lheckemann No particular reason. I'll change it to match.

**Comment by c0bw3b** [comment]
@GrahamcOfBorg build libgpgerror

**Comment by delroth** [comment]
I don't believe this change ever worked after you switched cp -> ln. The symlinks are broken because the directory name is repeated in the symlink source (compare the x86_64 command vs. aarch32 command).

**Pull request merged**